llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -20926,6 +20926,15 @@ RISCVTargetLowering::EmitInstrWithCustomInserter(MachineInstr &MI,
 
 void RISCVTargetLowering::AdjustInstrPostInstrSelection(MachineInstr &MI,
                                                         SDNode *Node) const {
+  // If instruction defines FRM operand, conservatively set it as non-dead to
+  // express data dependency with FRM users and prevent incorrect instruction
+  // reordering (these defs are marked as dead because at the moment of
+  // instruction emission they actually don't have any uses - they are added
+  // later in this hook).
```

Comment:
I don't think the code in InstrEmitter that controls the setting of the dead flag would see them even if they were done earlier. It only checks glued nodes or explicit data dependencies. Neither apply to the input case.